src/main.ts

```diff
@@ -129,7 +129,7 @@ export async function runAction(): Promise<any> {
         )
       )
     }
-    if (process.env.ACTIONS_RUNTIME_TOKEN) {
+    if (process.env.ACTIONS_RUNTIME_TOKEN && process.env.UPLOAD_ARTIFACTS) {
```

Comment:
You need to use core.getInput here too. You can assign the input to some variable and reuse that variable in both places, here and below. You can do something like `const uploadArtifacts = (core.getInput('upload-artifacts' === 'true');`